### PR TITLE
docs: replace torch.distributed.run by torchrun

### PR DESCRIPTION
**Pull request by panpan0000**
# What does this PR do?

The old way will be deprecated , move to new torchrun.
`FutureWarning: The module torch.distributed.launch is deprecated and will be removed in future. Use torchrun.`

This PR only addresses the doc part, not touching the unit-test to limit the impact.

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by panpan0000** [comment]
I just found an old PR which did the similar work  https://github.com/huggingface/transformers/pull/21780/files but out of sync.
but my PR just focus on DOC

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this! 

Just a few comments on removing the changes for the research examples. 

For any future readers who stumble on this PR: the previously closed PR requested [comments in the docs](https://github.com/huggingface/transformers/pull/21780#pullrequestreview-1315022415) for older pytorch versions. We now officially support pytorch >= 1.10. The entrypoint `torchrun` is present from [1.10 onwards](https://github.com/pytorch/pytorch/commit/65e6194aeb3269a182cfe2c05c122159da12770f).

File: examples/research_projects/distillation/README.md
[the remark this answers is not in the document]
Comment (panpan0000) [reply examples/research_projects/distillation/README.md:1]: got it . thank you @amyeroberts

**Comment by panpan0000** [comment]
> Thanks for adding this!
> 
> Just a few comments on removing the changes for the research examples.
> 
> For any future readers who stumble on this PR: the previously closed PR requested [comments in the docs](https://github.com/huggingface/transformers/pull/21780#pullrequestreview-1315022415) for older pytorch versions. We now officially support pytorch >= 1.10. The entrypoint `torchrun` is present from [1.10 onwards](https://github.com/pytorch/pytorch/commit/65e6194aeb3269a182cfe2c05c122159da12770f).

comments already addressed, now PR can be merged now.

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ def _inner_training_loop(
                 # references registered here no longer work on other gpus, breaking the module
                 raise ValueError(
                     "Currently --debug underflow_overflow is not supported under DP. Please use DDP"
-                    " (torch.distributed.launch)."
+                    " (torch.distributed.launch or torchrun)."
```
Comment (ArthurZucker) [inline src/transformers/trainer.py:1624]:
```suggestion
                    " (torchrun or torch.distributed.launch (deprecated))."
```

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**